### DISCUSSION TITLE
App Bridge v4.0

### DISCUSSION THREAD

**Discussion by pama1999** [Shopify/shopify-app-template-node#1318] [General]
Will there be an update to this app template where the newest app bridge will be added?

**Comment by transon-top** [comment]
I also want to ask this question because the Remix server framework is not very suitable for frequent CRUD operations in apps. Comparing the same logic in Remix and client-side rendering, it is obvious that client-side rendering provides a better user experience. So, I prefer to use node-template.

**Comment by AlexHooperDev** [comment]
Can anyone from Shopify chime in on this matter? We're at a point now where we're unable to take advantage of some recent features (i.e Managed Billing, App Bridge 4.0) because this repo isn't compatible. Our team are trying to decide whether it's time to migrate over to Remix or whether there'll be an incoming PR to this repo.

**Reply by matagaralph** [reply]
> Can anyone from Shopify chime in on this matter? We're at a point now where we're unable to take advantage of some recent features (i.e Managed Billing, App Bridge 4.0) because this repo isn't compatible. Our team are trying to decide whether it's time to migrate over to Remix or whether there'll be an incoming PR to this repo.

I created a fork of the project and updated it to use typescript. Now the template uses react router v7, typescript 5.7.*, vite 6 and Shopify Polaris to 13. Will make a pull request once i change the session storage to use prisma this week. I'm happy for me it paid off i submited the app and i didn't want to learn Remix.
